src/TitleDetails/TitleDetailsContainer.js

```diff
@@ -64,16 +105,61 @@ const TitleDetailsContainer = ({ location, history, mutator, match }) => {
     [history, title.id, location.search],
   );
 
+  const onAddPiece = useCallback(
+    (values) => {
+      const mutatorMethod = values.id ? 'PUT' : 'POST';
+
+      mutator.orderPieces[mutatorMethod](values)
+        .then(() => showCallout({
+          messageId: 'ui-receiving.piece.actions.addPiece.success',
+          type: 'success',
+          values: { title: title.title },
+        }))
+        .catch(() => showCallout({ messageId: 'ui-receiving.piece.actions.addPiece.error', type: 'error' }))
+        .finally(fetchPieces);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [fetchPieces, showCallout, title.title],
+  );
+
+  const onCheckIn = useCallback(
+    (values) => {
+      const mutatorMethod = values.id ? 'PUT' : 'POST';
+
+      mutator.orderPieces[mutatorMethod](values)
+        .then(piece => {
+          showCallout({
+            messageId: 'ui-receiving.piece.actions.addPiece.success',
+            type: 'success',
+            values: { title: title.title },
+          });
+
+          return checkInItems([{
+            ...piece,
+            itemStatus: ITEM_STATUS.inProcess,
+          }], mutator.checkIn);
+        })
+        .catch(() => showCallout({ messageId: 'ui-receiving.piece.actions.checkInItem.error', type: 'error' }))
+        .finally(fetchPieces);
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [fetchPieces, showCallout, title.title],
```

Comment:
please test this. usually `showCallout` causes issues in the dependencies array, since it always returns a new function

Reply:
I think it shouldn't return new function each time, and I don't see reason to include it here

Reply:
@NikitaSedyx  there is `useShowCallout()` called on every render, and inside there is return result of `something.bind()`. In case we improve this - we can specify it in dependency array, in other case it's better to not include this. Or am I wrong?